### DISCUSSION TITLE
Should ArgumentNullThrowHelper.ThrowIfNull be used instead of ArgumentNullException within AuthenticationState?

### DISCUSSION THREAD

**Discussion by aligoren** [dotnet/aspnetcore#51731] [General] [answered]
Hello, I hope the topic I mentioned is open in the right place.

In this file, last updated 2 years ago:

https://github.com/dotnet/aspnetcore/blob/main/src/Components/Authorization/src/AuthenticationState.cs#L19, there is the following statement:

```csharp
User = user ?? throw new ArgumentNullException(nameof(user));
```

Can we use this instead?

```csharp
ArgumentNullThrowHelper.ThrowIfNull(user);
User = user;
```

If this would be a good first contribution, I would like to send my very first MR :P

I hope I haven't disturbed here unnecessarily and for an irrelevant matter.

Best regards

**Comment by danmoseley** [comment, marked as answer]
Hi @aligoren you are in the right place and are welcome here. Many people contribute from all over the world and we would be happy for you to as well.

Yes we are interested in such changes. @divyeshio  began a change but was not able to complete it : https://github.com/dotnet/aspnetcore/pull/50666

You are welcome to offer a PR. My suggestion is to start small such as a PR for just one assembly. Small PRs are much easier to review and merge quickly. You could start with part of the above PR.

**Reply by danmoseley** [reply]
@BrennanConroy  any reason to believe that this change will need perf measurements? It should make the hot path shorter. Unless there is some odd interaction with inlining heuristics in some particular case.

**Reply by aligoren** [reply]
Hi @danmoseley I have opened a PR. Thank you very much for everything.

https://github.com/dotnet/aspnetcore/pull/51732

**Reply by BrennanConroy** [reply]
I don't think this needs any specific measurements. Like you said, this should make the 'hot path' shorter due to optimizations made when `throw` isn't in the main method body. But I don't think constructing `AuthenticationState` is on any hot path we're paying attention to right now 😄

**Reply by danmoseley** [reply]
Yes, haha, I was wondering more about the general case and what threshold we use for running TE measurements for example.

For cold paths it is arguably still a readability improvement.

**Comment by aligoren** [comment]
> Hi @aligoren you are in the right place and are welcome here. Many people contribute from all over the world and we would be happy for you to as well.
> 
> Yes we are interested in such changes. @divyeshio began a change but was not able to complete it : #50666
> 
> You are welcome to offer a PR. My suggestion is to start small such as a PR for just one assembly. Small PRs are much easier to review and merge quickly. You could start with part of the above PR.

Hey! Thank you so much for being kind to me. It really made me happy, and I'm sure it can encourage many others who might be hesitant just like me.

If I understood correctly, you suggested that I continue with the merge request mentioned above.

I should continue with this issue again, shouldn't I?

https://github.com/dotnet/aspnetcore/issues/50665

**Reply by danmoseley** [reply]
Yes please (but cc'ing @divyeshio  just so they're aware in case they are planning to start again)

As mentioned I suggest making a relatively small change and getting that merged first.
